.github/workflows/docs.yml

```diff
@@ -0,0 +1,187 @@
+name: Docs
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "docs/**"
+      - "solidity/**"
+      - "!solidity/dashboard/**"
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  docs-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event_name == 'pull_request' 
+      - uses: dorny/paths-filter@v2
+        if: github.event_name == 'pull_request' 
+        id: filter
+        with:
+          filters: |
+            path-filter:
+              - './((docs|(solidity/!dashboard))/**)'
+
+  docs-tex:
+    runs-on: ubuntu-latest
+    container: keepnetwork/texlive:3
+    needs: docs-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    defaults:
+      run:
+        working-directory: ./docs
+    env:
+      DOCS_DIR: ${{ github.workspace }}/tmp/docs
+    steps:
+      - uses: actions/checkout@v2
+
+      # Step generates files based on Makefile config. Current config produces
+      # a number of .png files, which are then used to generate 
+      # relay-states.pdf. We want to publish only the resulting .pdf file.
+      - name: Generate TeX document files
+        run: |
+          make clean
+          make docs
+          mkdir -p $DOCS_DIR
+          mv relay-states.pdf $DOCS_DIR
+
+      # A push event is a master merge; deploy to primary bucket.
+      - if: github.event_name == 'push'
+        name: Upload generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: cfc-production
+          bucket-name: docs.keep.network
+          bucket-path: core
+          build-folder: tmp/docs
+
+      # A pull_request event is a PR; deploy to preview bucket.
+      - if: github.event_name == 'pull_request'
+        name: Upload preview of generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: cfc-production
+          bucket-name: docs.keep.network
+          bucket-path: core/${{ github.head_ref }}
```

Comment:
Similar to above, using `${{ github.head_ref }}` is perfect here.

The “correct” way to do this per other builds is to have a `preview.docs.keep.network/<branch>/<everything else>`. That's actually not straightforward, because securing that with HTTPS requires some certificate wrangling that probably isn't worth doing right now.